prems-resources/backend/src/main/java/io/uhndata/cards/prems/internal/importer/AbstractConditionalClarityDataProcessor.java

```diff
@@ -0,0 +1,182 @@
+        EMPTY("is empty", true, (input, value) -> input == null || input.length() == 0),
+        NOT_EMPTY("is not empty", true, (input, value) -> input != null && input.length() > 0);
+
+        private final String operator;
+
+        private final boolean unary;
+
+        private final BiFunction<String, String, Boolean> evaluator;
+
+        Operator(final String operator, final boolean unary, final BiFunction<String, String, Boolean> evaluator)
+        {
+            this.operator = operator;
+            this.unary = unary;
+            this.evaluator = evaluator;
+        }
+
+        Pair<String, String> parse(String configuration)
+        {
+            if (this.unary) {
+                if (configuration.endsWith(this.operator)) {
+                    return Pair.of(StringUtils.removeEnd(configuration, this.operator).trim(), null);
+                }
+            } else {
+                String[] pieces = configuration.split("\\s+" + this.operator + "\\s+", 2);
+                if (pieces.length == 2) {
+                    return Pair.of(pieces[0], pieces[1]);
+                }
+            }
+            return null;
+        }
+
+        boolean matches(String column, String value)
+        {
+            return this.evaluator.apply(column, value);
+        }
+
+        private static boolean compareDouble(String input, String value,
+            BiFunction<Double, Double, Boolean> comparator)
+        {
+            try {
+                double inputNumber = Double.parseDouble(input);
+                double compareTo = Double.parseDouble(value);
+                return comparator.apply(inputNumber, compareTo);
+            } catch (Exception e) {
+                // Error parsing to double, does not match
+                return false;
+            }
+        }
+    }
+
+    private class ConditionDefinition
+    {
+        private final String column;
+
+        private final Operator operator;
+
+        private final String value;
+
+        ConditionDefinition(String configuration)
+        {
+            for (Operator o : Operator.values()) {
+                Pair<String, String> result = o.parse(configuration);
+                if (result != null) {
+                    this.column = result.getKey();
+                    this.operator = o;
+                    this.value = result.getValue();
+                    return;
+                }
+            }
+            this.column = null;
+            this.operator = null;
+            this.value = null;
+        }
+
+        public boolean isValid()
+        {
+            return this.column != null;
+        }
+
+        public boolean matches(Map<String, String> input)
+        {
+            return this.operator.matches(input.get(this.column), this.value);
+        }
+
+        @Override
+        public String toString()
+        {
+            return this.column + " " + this.operator + (this.value != null ? " " + this.value : "");
```

Comment:
With spaces around the operator names this can be simplified as:

```suggestion
            return this.column + this.operator + (this.value != null ? this.value : "");
```